### PR TITLE
SDIT-1233 Add method to convert NOMIS exclusions to DPS exclusions

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationService.kt [hunk 1]
```diff
@@ -146,6 +148,28 @@ class AllocationsMigrationService(
           null,
         )
       }
+
+  fun toSlots(exclusions: List<AllocationExclusion>): List<Slot> =
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationService.kt:152]: Not plumbed in yet, just wanted to get the conversion from NOMIS to DPS working first.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationService.kt [hunk 2]
```diff
@@ -146,6 +155,28 @@ class AllocationsMigrationService(
...
+    listOf("AM", "PM", "ED").map { timeSlot ->
+      exclusions
+        .filter { exclusion -> exclusion.slot == null || exclusion.slot.value == timeSlot }
+        .map { exclusion -> Slot.DaysOfWeek.entries.first { day -> day.value.startsWith(exclusion.day.value) } }
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationService.kt:163]: Maybe extract a method for readability?

**Review by andymarke — DISMISSED** [review]
Lgtm

**Review by andymarke — APPROVED** [review]
Lgtm

**Pull request merged**